pkg/tidydns/tidydns.go

```diff
@@ -84,15 +112,248 @@ const (
+	var userAllowFormatted []string
+	if len(userAllow) > 0 {
+		userAllowFormatted = make([]string, 0, len(userAllowFormatted))
+		for _, id := range userAllow {
+			userAllowFormatted = append(userAllowFormatted, strconv.Itoa(int(id)))
+		}
+	} else {
+		userAllowFormatted = []string{""}
+	}
+
+	var changePasswordOnFirstLoginFormatted string
+	if changePasswordOnFirstLogin {
+		changePasswordOnFirstLoginFormatted = "1"
+	} else {
+		changePasswordOnFirstLoginFormatted = "0"
+	}
+
+	data := url.Values{
+		"username":                       {username},
+		"epassword":                      {password},
+		"epassword_verify":               {password},
+		"change_password_on_first_login": {changePasswordOnFirstLoginFormatted},
+		"description":                    {description},
+		"auth_group":                     {strconv.Itoa(int(authGroup))},
+		//"tmp_auth_group":                 {""},
+		"user_allow": userAllowFormatted,
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", fmt.Sprintf("%s/=/user/new", c.baseURL), strings.NewReader(data.Encode()))
+	if err != nil {
+		return 0, err
+	}
+	req.SetBasicAuth(c.username, c.password)
+	req.Header.Set(headerContentType, mimeForm)
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return 0, err
+	}
+	defer func(Body io.ReadCloser) {
+		_ = Body.Close()
+	}(res.Body)
+	if res.StatusCode != http.StatusOK {
+		bodyBytes, err := io.ReadAll(res.Body)
+		bodyString := string(bodyBytes)
+		if err != nil {
+			return 0, err
+		}
+		if strings.Contains(bodyString, fmt.Sprintf("Key (username)=(%s) already exists", username)) {
+			return 0, fmt.Errorf("username already exists")
+		}
+		return 0, fmt.Errorf(errorTidyDNS, res.Status)
+	}
+
+	var user userCreate
+	err = json.NewDecoder(res.Body).Decode(&user)
+	if err != nil {
+		return 0, err
+	}
+
+	return UserID(user.Data.Id), nil
+}
+
+func (c *tidyDNSClient) GetInternalUser(ctx context.Context, userID UserID) (*UserInfo, error) {
+	req, err := http.NewRequestWithContext(
+		ctx,
+		"GET",
+		fmt.Sprintf("%s/=/user/%s", c.baseURL, strconv.Itoa(int(userID))),
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	req.SetBasicAuth(c.username, c.password)
+
+	res, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer func(Body io.ReadCloser) {
+		_ = Body.Close()
+	}(res.Body)
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(errorTidyDNS, res.Status)
+	}
+
+	var user userRead
+	err = json.NewDecoder(res.Body).Decode(&user)
+	if err != nil {
+		return nil, err
+	}
+
+	var groups = make([]UserInfoGroup, 0, len(user.Groups))
+	for _, group := range user.Groups {
+		groups = append(groups, UserInfoGroup{
+			GroupName:   group.GroupName,
+			Name:        group.Name,
+			Notes:       group.Notes,
+			Id:          group.Id,
+			Description: group.Description,
+		})
```

Comment:
should be possible to just convert the group, no?
```go
groups = append(groups, UserInfoGroup(group))
```
But like mentioned elsewhere, do we really need 2 types describing the same thing?

Reply:
Again, I would prefer to keep the public API and the internals separate to simply future refactoring

Reply:
But, only using the public API instead of both, changes nothing. Refactoring does not get harder, as you can always add the other internal struct back in the future should you need it. This only adds unnecessary complexity.

Reply:
https://en.wikipedia.org/wiki/KISS_principle

Reply:
Refactored in 9ab4bf49ad1f8ba99df183ded8c1d6f27a094fa3